sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -152,6 +169,17 @@ def __init__(self, urns, service):
   def provided_transforms(self):
     return self._urns.keys()
 
+  def with_underlying_provider(self, other_provider):
+    if not isinstance(other_provider, ExternalProvider):
+      return
+
+    all_urns = set(other_provider.schema_transforms().keys()).union(
```

Comment:
This line is part of the stack trace. I'm very confident this PR is the culprit, creating a revert for now since I'm not sure what the immediate solution is

Reply:
Hmm... the only thing red was coverage in the presubmits. 

Do you have a rollback already or should I create one?

Reply:
https://github.com/apache/beam/pull/31169 - still waiting on presubmits to go green

Reply:
I checked that we've published snapshots and everything though